roofit/roofitcore/src/RooProdPdf.cxx

```diff
@@ -594,7 +610,7 @@ void RooProdPdf::factorizeProduct(const RooArgSet& normSet, const RooArgSet& int
     RooArgSet pdfAllDeps; // All dependents of this PDF
 
     // Make list of all dependents of this PDF
-    RooArgSet* tmp = pdf->getObservables(normSet);
+    RooArgSet* tmp = getObservables(normSet);
```

Comment:
Try to never use raw pointers. Especially with arg sets ..., see #7890 and how cumbersome it is to find them.
Also, if you can, just put them on the stack.

Reply:
Done in later commit